dvc/objects/__init__.py

```diff
@@ -1,14 +1,28 @@
 import logging
 from concurrent.futures import ThreadPoolExecutor, as_completed
+from typing import TYPE_CHECKING, Optional
 
 from dvc.progress import Tqdm
 
+from .external import ExternalRepoFile
 from .tree import Tree
 
+if TYPE_CHECKING:
+    from .db.base import ObjectDB
+    from .file import HashFile
+
 logger = logging.getLogger(__name__)
 
 
-def save(odb, obj, jobs=None, **kwargs):
+def save(
+    odb: "ObjectDB",
+    obj: "HashFile",
+    jobs: Optional[int] = None,
+    **kwargs,
+):
+    if isinstance(obj, ExternalRepoFile):
+        obj = obj.get_staged(odb)
```

Comment:
I'm not sure if we will have more object types that work like erepo files in the future, but we could maybe name this `obj.resolve(odb)` instead, and just have it default to a no-op for regular `HashFiles` (to get rid of this erepo file specific isinstance() check)